src/Widgets/DisplayWidget.vala

```diff
@@ -136,24 +215,83 @@ public class Network.Widgets.DisplayWidget : Gtk.Grid {
                 return true;
             });
             break;
         case Network.State.FAILED_MOBILE:
-            image.icon_name = "network-cellular-offline-symbolic";
+            cellular_revealer.image.icon_name = "network-cellular-offline-symbolic";
+            cellular_connection_state = ConnectionState.DISCONNECTED;
+            update_icons ();
             break;
         case Network.State.FAILED_WIFI:
-        case Network.State.DISCONNECTED:
-            image.icon_name = "network-wireless-offline-symbolic";
+            wifi_revealer.image.icon_name = "network-wireless-offline-symbolic";
+            wifi_connection_state = ConnectionState.DISCONNECTED;
+            update_icons ();
             break;
+        case Network.State.FAILED_WIRED:
         case Network.State.WIRED_UNPLUGGED:
-            image.icon_name = "network-wired-offline-symbolic";
+            wired_revealer.image.icon_name = "network-wired-offline-symbolic";
+            wired_connection_state = ConnectionState.DISCONNECTED;
+            update_icons ();
             break;
         default:
-            image.icon_name = "network-offline-symbolic";
+            network_revealer.image.icon_name = "network-offline-symbolic";
+            cellular_connection_state = ConnectionState.DISCONNECTED;
+            wifi_connection_state = ConnectionState.DISCONNECTED;
+            wired_connection_state = ConnectionState.DISCONNECTED;
+            update_icons ();
             critical ("Unknown network state, cannot show the good icon: %s", state.to_string ());
             break;
         }
     }
+
+    private void update_icons () {
+        if ((cellular_connection_state + wifi_connection_state + wired_connection_state) > 0) {
+            network_revealer.reveal_child = false;
+        } else {
+            cellular_revealer.reveal_child = false;
+            wifi_revealer.reveal_child = false;
+            wired_revealer.reveal_child = false;
+            network_revealer.reveal_child = true;
+
+            return;
+        }
+
+        if (cellular_connection_state > 0) {
+            cellular_revealer.reveal_child = true;
+        } else {
+            cellular_revealer.reveal_child = false;
+        }
+
+        if (wifi_connection_state > 0) {
+            wifi_revealer.reveal_child = true;
+        } else {
+            wifi_revealer.reveal_child = false;
+        }
+
+        if (wired_connection_state > 0) {
+            wired_revealer.reveal_child = true;
+        } else {
+            wired_revealer.reveal_child = false;
+        }
+    }
+
+    private class ConnectionRevealer : Gtk.Revealer {
+        public Gtk.Image image { get; construct set; }
+        private string _icon_name;
+
+        public ConnectionRevealer.from_icon_name (string icon_name) {
+            _icon_name = icon_name;
+        }
+
+        construct {
+            image = new Gtk.Image.from_icon_name (_icon_name, Gtk.IconSize.LARGE_TOOLBAR);
```

Comment:
You probably need to add tooltips to the images themselves rather than the overall display widget.